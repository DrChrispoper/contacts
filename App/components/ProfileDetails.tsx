import React from 'react';
import { StyleSheet, TextStyle, ViewStyle, Text, View } from 'react-native';
import { Profile } from '../utils/types';

interface Style {
  item: ViewStyle;
  firstName: TextStyle;
  name: TextStyle;
  role: TextStyle;
  aboutTitle: TextStyle;
  description: TextStyle;
}

const styles = StyleSheet.create<Style>({
  item: {
    paddingRight: 20,
    paddingLeft: 20,
    marginVertical: 0,
    marginHorizontal: 0,
  },
  firstName: {
    fontWeight: 'bold',
  },
  name: {
    fontSize: 25,
    textAlign: 'center',
    fontWeight: '300',
  },
  role: {
    fontSize: 14,
    textAlign: 'center',
    color: '#808080',
  },
  aboutTitle: {
    paddingTop: 20,
    fontSize: 14,
    fontWeight: 'bold',
  },
  description: {
    fontSize: 14,
    color: '#808080',
    fontWeight: '300',
  },
});

interface Props {
  item: Profile;
  style?: ViewStyle;
  inDetailsPage?: boolean;
}

class ProfileDetails extends React.PureComponent<Props> {
  render(): JSX.Element {
    const { item, style, inDetailsPage = false } = this.props;
    return (
      <View style={[styles.item, style]}>
        {!inDetailsPage && (
          <View style={styles.name}>
            <Text style={styles.name}>
              <Text style={styles.firstName}>{item.firstName}</Text>
              {` ${item.lastName}`}
            </Text>
          </View>
        )}
        <Text style={styles.role}>{item.role}</Text>
        <Text style={styles.aboutTitle}>About me</Text>
        <Text style={styles.description}>{item.description}</Text>
      </View>
    );
  }
}

export default ProfileDetails;
