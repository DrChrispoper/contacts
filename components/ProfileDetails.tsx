import React from 'react';
import { StyleSheet, TextStyle, TouchableOpacity, ViewStyle, Text, View } from 'react-native';
import { Profile } from '../utils/types';

interface Style {
  item: ViewStyle;
  firstName: TextStyle;
  name: TextStyle;
  role: TextStyle;
  aboutTitle: TextStyle;
  description: TextStyle;
}

const styles = StyleSheet.create<Style>({
  item: {
    padding: 20,
    marginVertical: 8,
    marginHorizontal: 0,
  },
  firstName: {
    fontWeight: 'bold',
  },
  name: {
    fontSize: 25,
    textAlign: 'center',
    fontWeight: '300',
  },
  role: {
    fontSize: 14,
    textAlign: 'center',
    color: '#808080',
  },
  aboutTitle: {
    paddingTop: 20,
    fontSize: 14,
    fontWeight: 'bold',
  },
  description: {
    fontSize: 14,
    color: '#808080',
    fontWeight: '300',
  },
});

interface Props {
  item: Profile;
  onPress: () => void;
  style: ViewStyle;
}

const ProfileDetails: React.FC<Props> = ({ item, onPress, style }: Props) => {
  const names = item.name.split(' ');
  const firstName = names.shift();
  const lastName = names.join(' ');

  return (
    <TouchableOpacity onPress={onPress} style={[styles.item, style]}>
      <View style={styles.name}>
        <Text style={styles.name}>
          <Text style={styles.firstName}>{firstName}</Text>
          {` ${lastName}`}
        </Text>
      </View>
      <Text style={styles.role}>{item.role}</Text>
      <Text style={styles.aboutTitle}>About me</Text>
      <Text style={styles.description}>{item.description}</Text>
    </TouchableOpacity>
  );
};

export default ProfileDetails;
