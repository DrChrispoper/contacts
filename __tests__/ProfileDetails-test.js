import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from 'react-test-renderer';
import ProfileDetails from '../App/components/ProfileDetails';

test('renders correctly', () => {
  const profile = {
    id: 'DJKDSQLDQKS',
    fullName: 'Christopher Hockley',
    firstName: 'Christopher',
    lastName: 'Hockley',
    role: 'Developer',
    picture: jest.mock('../assets/images/avatars/Allan-Munger.png'),
    description:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.',
  };

  const tree = renderer.create(<ProfileDetails item={profile} style={{ height: 580 }} />).toJSON();
  expect(tree).toMatchSnapshot();
});
