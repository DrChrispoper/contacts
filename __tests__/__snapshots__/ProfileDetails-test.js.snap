// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "height": 580,
      "marginHorizontal": 0,
      "marginVertical": 0,
      "opacity": 1,
      "paddingLeft": 20,
      "paddingRight": 20,
    }
  }
>
  <View
    style={
      Object {
        "fontSize": 25,
        "fontWeight": "300",
        "textAlign": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 25,
          "fontWeight": "300",
          "textAlign": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        Christopher
      </Text>
       Hockley
    </Text>
  </View>
  <Text
    style={
      Object {
        "color": "#808080",
        "fontSize": 14,
        "textAlign": "center",
      }
    }
  >
    Developer
  </Text>
  <Text
    style={
      Object {
        "fontSize": 14,
        "fontWeight": "bold",
        "paddingTop": 20,
      }
    }
  >
    About me
  </Text>
  <Text
    style={
      Object {
        "color": "#808080",
        "fontSize": 14,
        "fontWeight": "300",
      }
    }
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
  </Text>
</View>
`;
